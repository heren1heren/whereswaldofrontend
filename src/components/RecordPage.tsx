import { FC } from 'react';
import { Layout } from './Layout';

import { RecordProps, recordType } from '../typeDeclaration';
import { useDeleteIncompleteRecords, useRecordFetch } from '../util';
export const Record: FC<RecordProps> = () => {
  const { errors: deleteErrors } = useDeleteIncompleteRecords(
    'http://localhost:3000/deleteIncompleteRecords'
  );

  const {
    isLoading,
    errors,
    data: records,
  } = useRecordFetch('http://localhost:3000/records');

  if (deleteErrors) return <div>{deleteErrors}</div>;
  if (errors) return <div>{errors}</div>;
  if (isLoading) return <div> ...Loading</div>;
  return (
    <Layout title="Record">
      {' '}
      <div className="table-responsive">
        <table className="table table-primary">
          <thead>
            <tr>
              <th scope="col">Rank</th>
              <th scope="col">Username</th>
              <th scope="col">Time(s)</th>
              <th scope="col"> Level</th>
            </tr>
          </thead>
          <tbody>
            {records ? (
              records.map((record: recordType, index: number) => {
                return (
                  <tr className="" key={Math.random()}>
                    <td scope="row">{index}</td>
                    <td>{record.username}</td>
                    <td>{record.duration}</td>
                    <td>{record.map}</td>
                  </tr>
                );
              })
            ) : (
              <div> empty record</div>
            )}
          </tbody>
        </table>
      </div>
    </Layout>
  );
};
