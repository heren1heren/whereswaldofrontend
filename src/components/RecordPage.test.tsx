import { render, screen } from '@testing-library/react';
import { Record } from './RecordPage';
import { MemoryRouter } from 'react-router-dom';
describe('initializing record page ', () => {
  // mock data from back end
  it('init', () => {
    const component = render(
      <MemoryRouter>
        <Record />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  // expect fetching correctly from backend
  // expect data to be display correctly
});
