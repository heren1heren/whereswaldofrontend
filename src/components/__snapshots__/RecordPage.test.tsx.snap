// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`initializing record page  > init 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" d-flex  flex-column justify-content-between vh-100 text-center"
      >
        <h1>
          Record
        </h1>
        <nav
          class="nav justify-content-center  "
        >
          <a
            class="nav-link "
            href="/records"
          >
             
            Record
          </a>
          <a
            class="nav-link "
            href="/"
          >
             
            Home Page
          </a>
        </nav>
        <main
          class=" flex-grow-1 p-lg-4"
        >
           
           
          <div
            class="table-responsive"
          >
            <table
              class="table table-primary"
            >
              <thead>
                <tr>
                  <th
                    scope="col"
                  >
                    Rank
                  </th>
                  <th
                    scope="col"
                  >
                    Username
                  </th>
                  <th
                    scope="col"
                  >
                    Time
                  </th>
                  <th
                    scope="col"
                  >
                     Level
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class=""
                >
                  <td
                    scope="row"
                  >
                    1
                  </td>
                  <td>
                    noname
                  </td>
                  <td>
                    00:00:00
                  </td>
                  <td>
                    The Space
                  </td>
                </tr>
                <tr
                  class=""
                >
                  <td
                    scope="row"
                  >
                    2
                  </td>
                  <td>
                    Grass
                  </td>
                  <td>
                    00:00:01
                  </td>
                  <td>
                    The Maze
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </main>
        <footer
          class=" text-center bg-black text-white"
        >
           
          Made By
          <a
            class=" ms-md-1  text-decoration-none"
            href="https://github.com/heren1heren/whereswaldofrontend"
          >
            Cookie Turtle
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" d-flex  flex-column justify-content-between vh-100 text-center"
    >
      <h1>
        Record
      </h1>
      <nav
        class="nav justify-content-center  "
      >
        <a
          class="nav-link "
          href="/records"
        >
           
          Record
        </a>
        <a
          class="nav-link "
          href="/"
        >
           
          Home Page
        </a>
      </nav>
      <main
        class=" flex-grow-1 p-lg-4"
      >
         
         
        <div
          class="table-responsive"
        >
          <table
            class="table table-primary"
          >
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  Rank
                </th>
                <th
                  scope="col"
                >
                  Username
                </th>
                <th
                  scope="col"
                >
                  Time
                </th>
                <th
                  scope="col"
                >
                   Level
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class=""
              >
                <td
                  scope="row"
                >
                  1
                </td>
                <td>
                  noname
                </td>
                <td>
                  00:00:00
                </td>
                <td>
                  The Space
                </td>
              </tr>
              <tr
                class=""
              >
                <td
                  scope="row"
                >
                  2
                </td>
                <td>
                  Grass
                </td>
                <td>
                  00:00:01
                </td>
                <td>
                  The Maze
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </main>
      <footer
        class=" text-center bg-black text-white"
      >
         
        Made By
        <a
          class=" ms-md-1  text-decoration-none"
          href="https://github.com/heren1heren/whereswaldofrontend"
        >
          Cookie Turtle
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
